hybrasyl/Objects/Creature.cs

```diff
@@ -330,42 +702,93 @@ public Creature LastHitter
         public bool PhysicalImmortal { get; set; }
         public bool MagicalImmortal { get; set; }
 
-        public virtual void Attack(Direction direction, Castable castObject, Creature target = null)
-        {
-            //do something?
-        }
 
-        public virtual void Attack(Castable castObject, Creature target)
+        public virtual bool UseCastable(Castable castObject, Creature target = null)
         {
-            //do spell?
-        }
+            // TODO: DRY if possible?
+            if (!Condition.CastingAllowed) return false;
+            var damage = castObject.Effects.Damage;
+            List<Creature> targets;
+            List<Creature> actualtargets;
+
+            if (target != null)
+                targets = new List<Creature> { target };
+            else
+                targets = GetTargets(castObject);
 
-        public virtual void Attack(Castable castObject)
-        {
-            //do aoe?
-        }
+            if (castObject.Effects?.Damage != null)
+                actualtargets = targets.Where(e => e is Monster || e.Condition.PvpEnabled).ToList();
+            else if (castObject.Effects?.Heal != null)
+                actualtargets = targets.Where(e => (e is Monster) == false).ToList();
+            else
+                actualtargets = targets; // yech
+
+            if (actualtargets.Count() == 0) return false;
+
+            foreach (var tar in actualtargets)
+            {
+                if (tar is Monster || (tar is User && ((User)tar).Condition.PvpEnabled))
+                {
+                    if (castObject.Effects?.Damage != null)
+                    {
+                        var damageOutput = NumberCruncher.CalculateDamage(castObject, tar, this);
+
+                        tar.Damage(damageOutput.Amount, OffensiveElement, damageOutput.Type, this);
+                    }
+                    else if (castObject.Effects?.Heal != null)
+                    {
+                        var healOutput = NumberCruncher.CalculateHeal(castObject, tar, this);
+                        tar.Heal(healOutput, this);
+                    }
+                    else if (castObject.Statuses?.Add != null)
+                    {
+
+                    }
+                    else if (castObject.Statuses?.Remove != null)
+                    {
+
```

Comment:
??

Reply:
This is a consequence of having this defined in two places by necessity (monster versus user) and working mostly on the user - I will refactor it a bit and update.

Reply:
👍